components/raftstore/src/store/fsm/peer.rs

```diff
@@ -429,8 +430,20 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
                 PeerMsg::ApplyRes { res } => {
                     self.on_apply_res(res);
                 }
-                PeerMsg::SignificantMsg(msg) => self.on_significant_msg(msg),
-                PeerMsg::CasualMessage(msg) => self.on_casual_msg(msg),
+                PeerMsg::SignificantMsg(msg) => {
+                    let timer = TiInstant::now_coarse();
+                    self.on_significant_msg(msg);
+                    RAFT_EVENT_DURATION
```

Comment:
Why observe them? Any messages that can take long time to finish?

Reply:
on handling merge result, it may destroy peer which writes rocksdb. I think it won't be very often, so add metrics here is acceptable.

Reply:
There can be a lot of `Unreachable` and `StoreResolved` messages in a short time.

Reply:
okay, I'll delete it